lib/matplotlib/tests/test_constrainedlayout.py

```diff
@@ -678,3 +679,16 @@ def test_constrained_toggle():
         assert not fig.get_constrained_layout()
         fig.set_constrained_layout(True)
         assert fig.get_constrained_layout()
+
+
+def test_layout_leak():
+    # Make sure there aren't any cyclic references when using LayoutGrid
+    # GH #25853
+    fig = plt.figure(constrained_layout=True, figsize=(10, 10))
+    fig.add_subplot()
+    fig.draw_without_rendering()
+    plt.close("all")
+    gc.collect()
+    for obj in gc.get_objects():
+        if isinstance(obj, mpl._layoutgrid.LayoutGrid):
+            pytest.fail("LayoutGrid object is still alive")
```

Comment:
We could silence codecov by writing an assert
```suggestion
    assert not any(isinstance(obj, mpl._layoutgrid.LayoutGrid)
                   for obj in gc.get_objects())
```

Reply:
Good catch. I was just copying the existing test that had this same pattern, so I guess we could also update that other test in a follow-up PR.